scripts/analysis.py

```diff
@@ -0,0 +1,378 @@
+    agent = cur.execute("select * from agententry where spec like '%sink%'")
+
+    for ag in agent:
+        sink_id.append(ag[1])
+    return sink_id
+
+
+def get_waste_id(resource_array):
+    """ Gets waste id from a resource array
+
+    Paramters
+    ---------
+    resource_array: array
+        array fetched from the resource table.
+
+    Returns
+    -------
+    waste_id: array
+        array of qualId for waste
+    """
+
+    wasteid = []
+
+    # get all the wasteid
+    for res in resource_array:
+        wasteid.append(res[7])
+
+    return set(wasteid)
+
+
+def exec_string(array, search, whatwant):
+    """ Generates sqlite query command to select things an
+        inner join between resources and transactions.
+
+    Parmaters
+    ---------
+    array: array
+        array of criteria that generates command
+    search: str
+        where [search]
+        criteria for your search
+    whatwant: str
+        select [whatwant]
+        column (set of values) you want out.
+
+    Returns
+    -------
+    exec_str: str
+        sqlite query command.
+    """
+
+    exec_str = ('select ' + whatwant + ' from resources inner join transactions\
+                on transactions.resourceid = resources.resourceid where '
+                + str(search) + ' = ' + str(array[0]))
+
+    for ar in array[1:]:
+        exec_str += ' or ' + str(ar)
+
+    return exec_str
+
+
+def get_sum(array, column_index):
+    """ Returns sum of a column in an array
+
+    Paramters:
+    ---------
+    array: array
+        array that contains a column with numbers
+    column_index: int
+        index for the column to be summed
+    """
+    sum = 0
+    for ar in array:
+        sum += ar[column_index]
+
+    return sum
+
+
+
+def isotope_calc(wasteid_array, snf_inventory, cursor):
+    """ Calculates isotope mass using mass fraction in compositions table.
+
+        Fetches all from compositions table.
+        Compositions table has the following format:
+            SimId / QualId / NucId / MassFrac
+        Then sees if the qualid matches, and if it multiplies
+        the mass fraction by the snf_inventory.
+
+    Prameters
+    ---------
+    wasteid_array: array
+        array of qualid of wastes
+    snf_inventory: float
+        total mass of snf
+    cursor: cursor
+        cursor for sqlite3
+
+    Returns
+    -------
+    nuclide_inven: array
+        inventory of individual nuclides.
+    """
+
+    # Get compositions of different commodities
+    # SimId / QualId / NucId / MassFrac
+    cur = cursor
+    comp = cur.execute('select * from compositions').fetchall()
+    total_snf_mass = get_sum(snf_inventory, 0)
+
+    nuclide_inven = 'total snf inventory = ' + str(total_snf_mass) + 'kg \n'
+    nuclides = []
+    mass_of_nuclides = []
+    # if the 'qualid's match,
+    # the nuclide quantity and calculated and displayed.
+    for isotope in comp:
+        for num in snf_inventory:
+            # num[1] = snf inventory qualid 
```

Comment:
why waste the lines on comments but not on renaming the unreadable variable? I recommend instead to have a line that says:

```python
inv_qualid = num[1]
```